src/MahApps.Metro/MahApps.Metro.Shared/Controls/Helper/TextBoxHelper.cs

```diff
@@ -247,24 +252,54 @@ private static void UseSpellCheckContextMenuChanged(DependencyObject d, Dependen
                 // set the spell check to true
                 tb.SetValue(SpellCheck.IsEnabledProperty, true);
                 // override pre defined context menu
-                tb.ContextMenu = GetDefaultTextBoxBaseContextMenu();
+                tb.ContextMenu = GetDefaultTextBoxBaseContextMenu(tb);
                 tb.ContextMenuOpening += TextBoxBaseContextMenuOpening;
             }
             else
             {
                 tb.SetValue(SpellCheck.IsEnabledProperty, false);
-                tb.ContextMenu = GetDefaultTextBoxBaseContextMenu();
+                tb.ContextMenu = GetDefaultTextBoxBaseContextMenu(tb);
                 tb.ContextMenuOpening -= TextBoxBaseContextMenuOpening;
             }
         }
 
+        private static void IsCutCopyPasteInContextMenuChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            var tb = d as TextBoxBase;
+            if (null == tb)
+            {
+                throw new InvalidOperationException("The property 'IsCutCopyPasteInContextMenuChanged' may only be set on TextBoxBase elements.");
+            }
+
+            tb.SetValue(IsCutCopyPasteInContextMenuProperty, e.NewValue);
+            tb.ContextMenu = GetDefaultTextBoxBaseContextMenu(tb);
+            AddExtraItemsToContextMenu(tb);
+            tb.ContextMenu.Visibility = tb.ContextMenu.Items.Count > 0 ? Visibility.Visible : Visibility.Hidden;
+        }
+
+        private static void ExtraContextMenuItemsChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            var tb = d as TextBoxBase;
+            if (null == tb)
+            {
+                throw new InvalidOperationException("The property 'ExtraContextMenuItemsChanged' may only be set on TextBoxBase elements.");
+            }
+            else
+            {
+                tb.SetValue(ExtraContextMenuItemsProperty, e.NewValue);
+                tb.ContextMenu = GetDefaultTextBoxBaseContextMenu(tb);
```

Comment:
Updating the context menu in each and every `XxxChanged` method seems wrong to me. I'd think that creating the actual context menu inside `TextBoxBaseContextMenuOpening` should be sufficient. @punker76 ?